catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipDecompression.java

```diff
@@ -0,0 +1,154 @@
+
+    private ZipValidator zipValidator;
+
+    /**
+     * Transforms a Zip InputStream into a List of {@link Metacard}s.  This method expects there to be a
+     * filePath and fileName key-value pair passed in the arguments map.
+     *
+     * @param inputStream - the InputStream to transform
+     * @param arguments   - the arguments for the transformation ("filePath" and "fileName").
+     * @return the List of {@link Metacard}s produced from the transformation.
+     * @throws CatalogTransformerException when the transformation fails.
+     */
+    @Override
+    public List<Metacard> transform(InputStream inputStream, Map<String, Serializable> arguments)
+            throws CatalogTransformerException {
+
+        if (inputStream == null) {
+            throw new CatalogTransformerException(
+                    "Unable to transform InputStream : InputStream was null.");
+        }
+
+        if (MapUtils.isEmpty(arguments) || !arguments.containsKey(FILE_PATH)
+                || !arguments.containsKey(FILE_NAME)) {
+            throw new CatalogTransformerException(
+                    "Unable to transform InputStream : Invalid arguments passed.");
+        }
+
+        String zipFileName = (String) arguments.get(FILE_PATH);
+        String zipPath = zipFileName + arguments.get(FILE_NAME);
+
+        try {
+            zipValidator.validateZipFile(zipPath);
+        } catch (ZipValidationException e) {
+            throw new CatalogTransformerException("Unable to validate Zip InputStream.  Ensure that the zip file is signed properly using jarsigner and the appropriate certificates.");
+        }
+
+        Map<String, Metacard> metacards = decompressFile(inputStream, zipFileName);
+        return metacards.values()
+                .stream()
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private Map<String, Metacard> decompressFile(InputStream inputStream, String zipFileName)
+            throws CatalogTransformerException {
+        try (ZipInputStream zipInputStream = new ZipInputStream(inputStream)) {
+            Map<String, Metacard> metacardMap = new HashMap<>();
+            ZipEntry zipEntry = zipInputStream.getNextEntry();
+
+            while (zipEntry != null) {
+                String filename = zipEntry.getName();
+
+                if (!filename.contains("META-INF")) {
+
+                    File zipEntryFile = new File(zipFileName + filename);
+
+                    if (!new File(zipEntryFile.getParent()).mkdirs()) {
+                        LOGGER.debug("File directory already exists in {}", zipFileName);
+                    }
+
+                    if (!zipEntryFile.isDirectory()) {
+                        FileOutputStream fileOutputStream = new FileOutputStream(zipEntryFile);
+
+                        int bytesRead;
+                        byte[] bytes = new byte[BUFFER_SIZE];
+                        while ((bytesRead = zipInputStream.read(bytes)) > 0) {
```

Comment:
`IOUtils` should work here (as opposed to my suggestion yesterday where you weren't actually copying bytes around).